### DISCUSSION TITLE
Running Console App in Browser

### DISCUSSION THREAD

**Discussion by fdmomtaz** [ZacharyPatten/dotnet-console-games#96] [General]
Hi, 

Let me start by saying this is a cool repo!

I have a simple and cool [console app](https://github.com/fdmomtaz/FancyHackerTerminal) and would like to run it in the browser similar to what you are doing with the "Play Now" buttons. I looked through the repo but couldn't find the answer on how to do this. I would appreciate any pointers on how to deploy the console app to run on the browser.

Thanks

**Comment by ZacharyPatten** [comment]
Howdy @fdmomtaz,

The "Play Now" buttons from the `README.md` link to GitHub Pages for which the source is set to the `github-pages` branch in this repository. When the `main` branch is pushed to there is a GitHub Action, [`Website Deploy.yml`](https://github.com/dotnet/dotnet-console-games/blob/main/.github/workflows/Website%20Deploy.yml), that builds the [`Website`](https://github.com/dotnet/dotnet-console-games/tree/main/Projects/Website) project and force pushes the build output into the `github-pages` branch. [`Website`](https://github.com/dotnet/dotnet-console-games/tree/main/Projects/Website) is a Blazor Webassembly project that contains ports of all the console games. The process of porting the code from Console Application to Blazor Webassembly is currently a manual process I perform, although making a code generator is definitely possible; see https://github.com/dotnet/dotnet-console-games/issues/23. In the [`Website`](https://github.com/dotnet/dotnet-console-games/tree/main/Projects/Website) project there is [BlazorConsole.cs](https://github.com/dotnet/dotnet-console-games/blob/main/Projects/Website/BlazorConsole.cs) that contains reimplementations of most of the members of `System.Console` to work in Blazor Webassembly so the majority of the porting process is a simple copy and paste.

**TLDR: see the [`Website`](https://github.com/dotnet/dotnet-console-games/tree/main/Projects/Website) project for the source code of the web implementations**

Hopefully that helps. :) Feel free to ask additional questions.
